protocol/shannon/websocket_context.go

```diff
@@ -44,21 +44,114 @@ type websocketRequestContext struct {
 	selectedEndpoint endpoint
 }
 
+// ---------- Websocket Request Context Setup  ----------
+
+// BuildWebsocketRequestContextForEndpoint creates a new WebSocket protocol request context for a specified service and endpoint.
+// This method immediately establishes the WebSocket connection and starts the bridge.
+//
+// Parameters:
+//   - ctx: Context for cancellation, deadlines, and logging.
+//   - serviceID: The unique identifier of the target service.
+//   - selectedEndpointAddr: The address of the endpoint to use for the request.
+//   - httpReq: HTTP request used for WebSocket upgrade and delegated mode app extraction.
+//   - httpResponseWriter: HTTP response writer for WebSocket upgrade.
+//   - messageObservationsChan: Channel for sending message-level observations to the gateway.
+func (p *Protocol) BuildWebsocketRequestContextForEndpoint(
+	ctx context.Context,
+	serviceID protocol.ServiceID,
+	selectedEndpointAddr protocol.EndpointAddr,
+	websocketMessageProcessor websockets.WebsocketMessageProcessor,
+	httpReq *http.Request,
+	httpResponseWriter http.ResponseWriter,
+	messageObservationsChan chan *observation.RequestResponseObservations,
```

Comment:
Please leave a TODO: this channel should be created here, not passed to it, as protocol is the producer side of the channel.